src/main/java/it/gov/pagopa/payment/service/payment/qrcode/QRCodePreAuthServiceImpl.java

```diff
@@ -96,36 +93,31 @@ public AuthPaymentDTO relateUser(String trxCode, String userId) {
 
   private void checkPreAuth(String trxCode, String userId, TransactionInProgress trx, String walletStatus) {
     if (PaymentConstants.WALLET_STATUS_SUSPENDED.equals(walletStatus)){
-      throw new ClientExceptionWithBody(
-              HttpStatus.FORBIDDEN,
+      throw new ForbiddenException(
```

Comment:
I think that creating a "wrap" of http exception could produce the same effect of `ClientExceptionWithBody`. Maybe I would create errors that have business significance. For example in this method `checkPreAuth` I would map the various exceptions into: `UserSuspendedException`, `TransactionNotFoundOrExpired`, `TransactionAlreadyAuthorized` and so on. Then I would use the same map that you created `transcodeMap` to map this exceptions into right HTTP Code.

Reply:
The aim is to provide a generic and centralized way to categorize the business exception, in order to simplify their translation into the presentation layer format, without introduce dependencies on application specific exception.

Anyway, in the particular application we could create extension of the generic exceptions in order to allow the creation and throw of exceptions such `UserSuspendedException`, which could be an application specific exception extending the `ForbiddenException`
It's translation into HTTP exception could continue to be generic as done in the current PR

Reply:
I think business exceptions cannot be centralized, as they depend on the responsibilities of the microservice. We can centralize the translation to presentation layer by handling the mapping to http, for example, the approach suggested with this PR by using a map for me is very effective. 
```java
Map.of(
      UserSuspendedException.class, HttpStatus.FORBIDDEN,
      ...
  );
  ```
  
  By injecting a map or a mapping function you can centralize the mapping to presentation layer while you describe errors in terms of business

Reply:
I understand your point: the solution proposed is throwing the "consequence" of the operation (eg: Forbidden) not its cause (eg: UserSuspended)

As you suggest, we could enforce the application to configure a Bean, `ServiceExceptionMap extend Map<Class<? extends ServiceException>, HttpStatus>`, where the `ServiceException` will be the abstract class provided in this PR, removing all the other subclasses, letting the centralized way to build and return the ErrorDTO

let's talk about this in the call scheduled this afternoon, but I'm quite convinced :)